vars/cloudFoundryDeploy.groovy

```diff
@@ -198,6 +204,7 @@ void call(Map parameters = [:]) {
             .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
             .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS, CONFIG_KEY_COMPATIBILITY)
             .mixin(parameters, PARAMETER_KEYS, CONFIG_KEY_COMPATIBILITY)
+            .addIfEmpty('deployTool', script.commonPipelineEnvironment.getBuildTool()=='mta' ? 'mtaDeployPlugin' : 'cf_native')
```

Comment:
This will now ignore if the user configured something in general.buildTool. 
I think you need the depending on mixin syntax